sharktank/tests/models/llama/benchmark-tests.py

```diff
@@ -0,0 +1,310 @@
+            f"--input=@{self.decode_args_fp8}/seq_lens.npy",
+            f"--input=@{self.decode_args_fp8}/start_positions.npy",
+            f"--input=@{self.decode_args_fp8}/seq_block_ids.npy",
+            f"--input=@{self.decode_args_fp8}/cache_state_f16.npy",
+            "--benchmark_repetitions=3",
+        ]
+
+    def testBenchmark8B_f16_Decomposed(self):
+        self.export_mlir(
+            "decomposed",
+            self.irpa_path,
+            self.output_mlir,
+            self.output_json,
+            self.repo_root,
+        )
+        self.iree_compile(
+            self.output_mlir, self.output_vmfb, self.iree_compile_args, self.repo_root
+        )
+        # benchmark prefill
+        self.iree_benchmark_module(
+            "0",
+            self.output_vmfb_fp8,
+            self.irpa_path_fp8,
+            self.iree_run_prefill_args_fp8,
+            self.repo_root,
+        )
+        # benchmark decode
+        self.iree_benchmark_module(
+            "0",
+            self.output_vmfb_fp8,
+            self.irpa_path_fp8,
+            self.iree_run_decode_args_fp8,
+            self.repo_root,
+        )
+
+    @pytest.mark.xfail
+    def testBenchmark8B_f16_Non_Decomposed(self):
+        self.export_mlir(
+            "torch_sdpa",
+            self.irpa_path,
+            self.output_mlir,
+            self.output_json,
+            self.repo_root,
+        )
+        self.iree_compile(
+            self.output_mlir, self.output_vmfb, self.iree_compile_args, self.repo_root
+        )
+        # benchmark prefill
+        self.iree_benchmark_module(
+            "0",
+            self.output_vmfb,
+            self.irpa_path,
+            self.iree_run_prefill_args,
+            self.repo_root,
+        )
+        # benchmark decode
+        self.iree_benchmark_module(
+            "0",
+            self.output_vmfb,
+            self.irpa_path,
+            self.iree_run_decode_args,
+            self.repo_root,
+        )
+
+    @pytest.mark.xfail
+    def testBenchmark8B_fp8_Decomposed(self):
+        self.export_mlir(
+            "decomposed",
+            self.irpa_path_fp8,
+            self.output_mlir_fp8,
+            self.output_json_fp8,
+            self.repo_root,
+        )
+        self.iree_compile(
+            self.output_mlir_fp8,
+            self.output_vmfb_fp8,
+            self.iree_compile_args,
+            self.repo_root,
+        )
+        # benchmark prefill
+        self.iree_benchmark_module(
+            "0",
+            self.output_vmfb,
+            self.irpa_path,
+            self.iree_run_prefill_args,
+            self.repo_root,
+        )
+        # benchmark decode
+        self.iree_benchmark_module(
+            "0",
+            self.output_vmfb,
+            self.irpa_path,
+            self.iree_run_decode_args,
+            self.repo_root,
+        )
+
+    @pytest.mark.xfail
+    def testBenchmark8B_fp8_Non_Decomposed(self):
```

Comment:
can you add TP1 70B and TP8 405B as well?

Reply:
Done